### PR TITLE
Fixes for Error Swallowing

### PR DESCRIPTION
**Pull request by scribe**
Three changes in this pull request:
Stop swallowing on failure from workers, mostly in DS3JobTask
Ensure logging takes place on setOnFailure. Searched for everywhere we setOnFailure and tried to make sure we log. All over the place
In order to make sure I didn't miss places I ended up replacing JobTask constructors with Guice Factories to simplify things.
One Liner: We now check settings correctly before setting the metadata receiver in Ds3GetJob

### CODE REVIEW

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java [hunk 1]
```diff
@@ -419,14 +419,15 @@ private void handleDropEvent(final DragEvent event, final TreeTableRow<Ds3TreeTa
...
+                    LOG.info("Put Job Succeed");
                     Ds3PanelService.refresh(selectedItem);
                     ds3TreeTable.getSelectionModel().clearSelection();
                     ds3TreeTable.getSelectionModel().select(selectedItem);
 
                 }));
                 putJob.setOnFailed(SafeHandler.logHandle(e -> {
-                    LOG.info("setOnFailed");
+                    LOG.info("Put Job Failed", e.getSource().getException());
+                    loggingService.logMessage("Put job failed with message: " + e.getSource().getException().getMessage(), LogType.ERROR);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java]: Extract the exception to a variable so you only get it once.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java [hunk 2]
```diff
@@ -258,6 +259,10 @@ protected Object call() {
                     ds3CancelSingleJobTask.setOnFailed(SafeHandler.logHandle(cancelJobTaskFailedEvent -> {
                         LOG.info("Cancellation of interrupted job " + jobId + " failed");
                     }));
+                    ds3CancelSingleJobTask.setOnFailed(SafeHandler.logHandle(event -> {
+                        LOG.error("Cancellation of recoved job faile");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java]: Seems like a typeo in here.  The text probably should read: "Cancellation of recovered job failed"

Also the exception is not being logged.  Get that from the event.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItem.java [hunk 3]
```diff
@@ -119,7 +119,7 @@ public FontAwesomeIconView getGraphicFont(final FileTreeModel fileTreeModel) {
                 super.setGraphic(previousGraphics);
             }));
             buildChildren.setOnFailed(SafeHandler.logHandle(event -> {
-                LOG.info("Success");
+                LOG.info("Failed to build children");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/FileTreeTableItem.java]: This should be an error log.  Also we're not logging the exception.  And we probably should describe more about what it means by 'children.'

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java [hunk 4]
```diff
@@ -359,10 +345,7 @@ private void transferToBlackPearl() {
             return;
         }
 
-        final String priority = !savedJobPrioritiesStore.getJobSettings().getPutJobPriority().equals(resourceBundle.getString("defaultPolicyText"))
-                ? savedJobPrioritiesStore.getJobSettings().getPutJobPriority()
-                : null;
-        startPutJob(session, filesToPut, bucket, targetDir, priority, jobInterruptionStore, remoteDestination);
+        startPutJob(session, filesToPut, bucket, targetDir, jobInterruptionStore, remoteDestination);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: I know you didn't write this code, but why is the initialize method side effecting?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java [hunk 5]
```diff
@@ -466,7 +449,8 @@ private void startGetJob(final List<Ds3TreeTableValueCustom> listFiles,
             refreshFileTreeView();
         }));
         getJob.setOnFailed(SafeHandler.logHandle(event -> {
-            LOG.error("Get Job failed");
+            LOG.error("Get Job failed", event.getSource().getException());
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: Extract the exception to a variable.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java [hunk 6]
```diff
@@ -487,25 +471,17 @@ private void startPutJob(final Session session,
...
         }));
         putJob.setOnFailed(SafeHandler.logHandle(failEvent -> {
-            final UUID jobId = putJob.getJobId();
-            if (jobId == null) {
-                LOG.info("BULK_PUT job Failed without receiving an ID");
-            } else {
-                LOG.info("BULK_PUT job {} Failed.", jobId);
-                refreshBlackPearlSideItem(remoteDestination);
-            }
+            LOG.error("Put Job Failed", failEvent.getSource().getException());
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: Extract the exception to a single variable.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3JobTask.java [hunk 7]
```diff
@@ -119,7 +113,7 @@ void getTransferRates(final Instant jobStartInstant, final AtomicLong totalSent,
                     totalJobSize, sourceLocation, targetLocation).toString());
         }
 
-        updateProgress(totalSent.get() / 2 , totalJobSize);
+        updateProgress(totalSent.get() / 2, totalJobSize);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3JobTask.java]: Do you know why we need to divide by two?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/RefreshCompleteViewWorker.java [hunk 8]
```diff
@@ -96,6 +96,7 @@ public static void refreshCompleteTreeTableView(final Ds3Common ds3Common, final
                     }));
                     getServiceTask.setOnFailed(SafeHandler.logHandle(event -> {
                         LOG.info("GetServiceTask failed");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/RefreshCompleteViewWorker.java]: This should log using LOG.error, and it should log the exception.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTableItem.java [hunk 9]
```diff
@@ -141,7 +145,11 @@ private void buildChildren(final ObservableList<TreeItem<Ds3TreeTableValue>> obs
                 ds3Common.getDs3TreeTableView().setPlaceholder(null);
         }));
         getBucketTask.setOnCancelled(SafeHandler.logHandle(event -> super.setGraphic(previousGraphics)));
-        getBucketTask.setOnFailed(SafeHandler.logHandle(event -> super.setGraphic(previousGraphics)));
+        getBucketTask.setOnFailed(SafeHandler.logHandle(event -> {
+            super.setGraphic(previousGraphics);
+            loggingService.logMessage("Failed to get bucket", LogType.ERROR);
+            LOG.error("Failed to get bucket");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTableItem.java]: This needs to log the exception that was thrown.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java [hunk 10]
```diff
@@ -419,14 +419,15 @@ private void handleDropEvent(final DragEvent event, final TreeTableRow<Ds3TreeTa
...
-                    LOG.info("Succeed");
+                    LOG.info("Put Job Succeed");
                     Ds3PanelService.refresh(selectedItem);
                     ds3TreeTable.getSelectionModel().clearSelection();
                     ds3TreeTable.getSelectionModel().select(selectedItem);
 
                 }));
                 putJob.setOnFailed(SafeHandler.logHandle(e -> {
-                    LOG.info("setOnFailed");
+                    LOG.info("Put Job Failed", e.getSource().getException());
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java]: Log using the error level.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java [hunk 11]
```diff
@@ -370,6 +376,8 @@ public void refresh(final TreeTableView<JobInfoModel> treeTableView, final JobIn
             treeTableView.setPlaceholder(new Label(resourceBundle.getString("dontHaveInterruptedJobs")));
         }));
         getJobIDs.setOnFailed(SafeHandler.logHandle(event -> {
+            LOG.error("Get Job IDs failed");
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java]: Make sure to log the exception.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3JobTask.java (hunk 7 above)
Comment (scribe) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3JobTask.java]: I have no clue, but it is done that way everywhere I'll look into it for a moment.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java (hunk 4 above)
Comment (scribe) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: It's a little obscure but this is inside a callback for setOnAction on a button.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java [hunk 12]
```diff
@@ -419,14 +419,16 @@ private void handleDropEvent(final DragEvent event, final TreeTableRow<Ds3TreeTa
...
                     Ds3PanelService.refresh(selectedItem);
                     ds3TreeTable.getSelectionModel().clearSelection();
                     ds3TreeTable.getSelectionModel().select(selectedItem);
 
                 }));
                 putJob.setOnFailed(SafeHandler.logHandle(e -> {
-                    LOG.info("setOnFailed");
+                    final Throwable exception = e.getSource().getException();
+                    LOG.error("Put Job Failed", exception);
+                    loggingService.logMessage("Put job failed with message: " + exception, LogType.ERROR);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java]: You should use `exception.getMessage()` in the `logMessage` message.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java [hunk 13]
```diff
@@ -241,10 +242,11 @@ protected Object call() {
                     refresh(buttonCell.getTreeTableView(), jobInterruptionStore, endpointInfo);
                 }));
                 recoverInterruptedJob.setOnFailed(SafeHandler.logHandle(recoverInterruptedJobFailedEvent -> {
+                    LOG.error("Failed to recover", recoverInterruptedJob.getException());
                     loggingService.logMessage("Failed to recover " + filesAndFolderMap.getType() + " job " + endpointInfo.getEndpoint(), LogType.ERROR);
                     refresh(buttonCell.getTreeTableView(), jobInterruptionStore, endpointInfo);
                 }));
-                recoverInterruptedJob.setOnCancelled(SafeHandler.logHandle(recoverInterruptedJobCancelledEvent -> {
+                recoverInterruptedJob.setOnCancelled(SafeHandler.logHandle((WorkerStateEvent recoverInterruptedJobCancelledEvent) -> {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java]: Is this cast really necessary?

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java [hunk 14]
```diff
@@ -255,8 +257,12 @@ protected Object call() {
                         LOG.info("Cancellation of recovered job success");
                         refresh(buttonCell.getTreeTableView(), jobInterruptionStore, endpointInfo);
                     }));
-                    ds3CancelSingleJobTask.setOnFailed(SafeHandler.logHandle(cancelJobTaskFailedEvent -> {
-                        LOG.info("Cancellation of interrupted job " + jobId + " failed");
+                    ds3CancelSingleJobTask.setOnFailed(SafeHandler.logHandle((WorkerStateEvent cancelJobTaskFailedEvent) -> {
+                        LOG.error("Cancellation of interrupted job " + jobId + " failed", cancelJobTaskFailedEvent.getSource().getException());
+                    }));
+                    ds3CancelSingleJobTask.setOnFailed(SafeHandler.logHandle((WorkerStateEvent event) -> {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java]: It looks like you're overwriting the setOnFailed called back.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java (hunk 4 above)
Comment (rpmoore) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java]: Oh I see it now.  Github was playing tricks on me.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3JobTask.java (hunk 7 above)
Comment (rpmoore) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3JobTask.java]: If this is done everywhere it's used, let's pull it into the updateProgress method.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java (hunk 13 above)
Comment (scribe) [reply dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/interruptedjobwindow/JobInfoPresenter.java]: For some reason yes, we can't resolve getException without it

**Pull request merged**